39.JS_WebStorage/assets/js/main.js

```diff
@@ -0,0 +1,142 @@
+const API_KEY = "1ee08fee54c092fa533c1c8db5b1ff95";
+const WEATHER_API_URL = `http://api.openweathermap.org/data/3.0/onecall?appid=${API_KEY}`;
+const WEATHER_ICON_API_URL = `https://openweathermap.org/img/wn/`;
+
+const WEATHER_STORAGE_KEY = "weatherData";
+
+const citySelect = document.querySelector('select[name="city"]');
+const weatherContainer = document.querySelector(".weather-container");
+const detailsContainer = document.querySelector(".details-container");
+
+const saveToStorage = (key = WEATHER_STORAGE_KEY, data) => {
+  localStorage.setItem(key, JSON.stringify(data));
+};
+
+const getFromStorage = (key = WEATHER_STORAGE_KEY) => {
+  return JSON.parse(localStorage.getItem(key));
```

Comment:
pls add some condition in case if data will lost or in incorrect format in this case return empty object